### ISSUE TITLE
Issue with AzureBlob File Copy VSTS Task

### ISSUE THREAD

**Issue by Qrtschak** [Microsoft/azure-pipelines-tasks#8652]
## Environment
VSTS

- Agent - Hosted or Private:

    - If using private agent, provide the OS of the machine running the agent and the agent version: Windows Server 2016 Datacenter, Agent version 2.138.4

## Issue Description

The agent while realizing VSTS task AzureBlob File Copy, which is to copy some simple data to the Azure Blob, when choosing the subscription and tenantId is throwing the following error and stopping - logs below. Of course subscription ID and tenant ID have been checked they come from service connection definition in VSTS and are fine. This task works on other Agent on Azure VM, but we can't find the exact cause that it is not working on this and few other. Any ideas what is missing or misconfigured maybe Guys? Thanks a lot for Your help.

### Task logs

2018-10-17T14:09:54.9270831Z ##[section]Starting: AzureBlob File Copy
2018-10-17T14:09:54.9279148Z ==============================================================================
2018-10-17T14:09:54.9279688Z Task         : Azure File Copy
2018-10-17T14:09:54.9279945Z Description  : Copy files to Azure blob or VM(s)
2018-10-17T14:09:54.9280217Z Version      : 2.0.10
2018-10-17T14:09:54.9280440Z Author       : Microsoft Corporation
2018-10-17T14:09:54.9280722Z Help         : [More Information](https://aka.ms/azurefilecopyreadme)
2018-10-17T14:09:54.9281017Z ==============================================================================
2018-10-17T14:09:57.3948412Z Added TLS 1.2 in session.
2018-10-17T14:09:58.9352758Z ##[command]Import-Module -Name C:\Program Files\WindowsPowerShell\Modules\AzureRM\6.2.1\AzureRM.psd1 -Global
2018-10-17T14:11:20.0628248Z ##[command]Clear-AzureRmContext -Scope Process
2018-10-17T14:11:21.7997012Z ##[command]Clear-AzureRmContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue
2018-10-17T14:11:22.7431537Z ##[command]Add-AzureRMAccount -ServicePrincipal -Tenant *** -Credential System.Management.Automation.PSCredential -Environment AzureCloud
2018-10-17T14:11:24.4480620Z ##[command] Select-AzureRMSubscription -SubscriptionId *** -TenantId ***
2018-10-17T14:11:26.0997150Z ##[error]Please provide a valid tenant or a valid subscription.
2018-10-17T14:11:26.2478666Z ##[section]Finishing: AzureBlob File Copy

### Error logs

2018-10-17T14:11:24.4480620Z ##[command] Select-AzureRMSubscription -SubscriptionId *** -TenantId ***
2018-10-17T14:11:26.0997150Z ##[error]Please provide a valid tenant or a valid subscription.
2018-10-17T14:11:26.2478666Z ##[section]Finishing: AzureBlob File Copy

**Comment by Ajay-MS** [comment]
@Qrtschak 

Is this issue consistent or intermittent?

This issue occurs when `Add-AzureRMAccount` cmdlet returns an empty list of account.  Hence `Select-AzureRMSubscription` cmdlet get failed. 

One of a possible reason is AAD app is not being assigned to the subscription. But as you mentioned endpoint is working fine on other machines. I am ruling out this option. 

If the above issue is consistent then please run the below-mentioned script on agent machine:
```
$servicePrincipalId = "<ServicePrincipalID>"
$servicePrincipalKey = "<servicePrincipalKey>"
$tenantId = "<tenantId>"
$azureSubscriptionId = "<azureSubscriptionId>" 

$securePassword = ConvertTo-SecureString $servicePrincipalKey -AsPlainText -Force
$psCredential = New-Object System.Management.Automation.PSCredential ($servicePrincipalId, $securePassword)

Write-Verbose "Add-AzureRMAccount Started" -Verbose
$azureAccount = Add-AzureRMAccount -ServicePrincipal -Tenant $tenantId -Credential $psCredential -EnvironmentName AzureCloud
$azureAccount

Get-AzureRMContext
```

Also run Azure File Copy task again by enabling debug logging. Debug logging can be enabled by adding variable `system.debug` with value `true` in release defintion. 

Please share output of script and debug logs to RM_Customer_Queries@microsoft.com. 

Also, I have seen similar context releated issues with Azure PS version less than 6.5 . I will suggest you to tryout with Azure PS version 6.7 or greater and let me know if it works for you.

**Comment by Ajay-MS** [comment]
@Qrtschak 

Is this issue consistent or intermittent?

This issue occurs when `Add-AzureRMAccount` cmdlet returns an empty list of account.  Hence `Select-AzureRMSubscription` cmdlet get failed. 

One of a possible reason is AAD app is not being assigned to the subscription. But as you mentioned endpoint is working fine on other machines. I am ruling out this option. 

If the above issue is consistent then please run the below-mentioned script on agent machine:
```
$servicePrincipalId = "<ServicePrincipalID>"
$servicePrincipalKey = "<servicePrincipalKey>"
$tenantId = "<tenantId>"
$azureSubscriptionId = "<azureSubscriptionId>" 

$securePassword = ConvertTo-SecureString $servicePrincipalKey -AsPlainText -Force
$psCredential = New-Object System.Management.Automation.PSCredential ($servicePrincipalId, $securePassword)

Write-Verbose "Add-AzureRMAccount Started" -Verbose
$azureAccount = Add-AzureRMAccount -ServicePrincipal -Tenant $tenantId -Credential $psCredential -EnvironmentName AzureCloud
$azureAccount

Get-AzureRMContext
```

Also run Azure File Copy task again by enabling debug logging. Debug logging can be enabled by adding variable `system.debug` with value `true` in release definition. 

Please share the output of the script and debug logs to RM_Customer_Queries@microsoft.com. 

Also, I have seen similar context related issues with Azure PS version less than 6.5. I will suggest you to try out with Azure PS version 6.7 or greater and let me know if it works for you.

**Comment by Ajay-MS** [comment]
@Qrtschak 

Thanks for the confirmation that issue has been resolved for you after upgrading to Azure PS 6.11
